src/main/java/com/magicvet/Main.java

```diff
@@ -59,24 +59,19 @@ static void registerNewClient(){
     static Client buildClient(String email) {
         Client client = new Client();
         client.email = email;
+        client.firstName = checkInputName("First name");
+        client.lastName = checkInputName("Last name");;
+        return client;
+    }
 
-        System.out.print("First name: ");
-        String firstName = SCANNER.nextLine();
-        while (!isNameValid(firstName)) {
-            System.out.println("The first name is not valid! Please provide the valid name (only latin letters and hyphen are allowed, the name should be at least 3 letters long): ");
-            firstName = SCANNER.nextLine();
-        }
-        client.firstName = firstName;
-
-        System.out.print("Last name: ");
-        String lastName = SCANNER.nextLine();
-        while (!isNameValid(lastName)) {
-            System.out.println("The last name is not valid! Please provide the valid name (only latin letters and hyphen are allowed, the name should be at least 3 letters long): ");
-            lastName = SCANNER.nextLine();
+    static String checkInputName(String inputPrompt) {
+        System.out.print(inputPrompt + ": ");
+        String name = SCANNER.nextLine();
+        while (!isNameValid(name)) {
+            System.out.println("The " + inputPrompt +" is not valid! Please provide the valid name (only latin letters and hyphen are allowed, the name should be at least 3 letters long): ");
+            name = SCANNER.nextLine();
         }
```

Comment:
It is a great idea to extract the logic for getting and validating the first and last names into a separate method. This approach makes the code more modular and avoids duplication by reusing the same validation logic for first and last names.